indexing-service/src/main/java/io/druid/indexing/overlord/autoscaling/ScalingStats.java

```diff
@@ -65,6 +65,15 @@ public ScalingStats(int capacity)
     }
   }
 
+  public ScalingStats(ScalingStats stats1, ScalingStats stats2)
+  {
+    this(stats1.recentEvents.size() + stats2.recentEvents.size());
+    synchronized (lock) {
```

Comment:
Refactored in `ScalingStats.addAll()` and no new constructor.

@egor-ryashin capacity=0 is handled in the existing constructor of `ScalingStats`